packages/protobuf/src/google/protobuf/duration_pb.ts

```diff
@@ -131,7 +130,7 @@ export class Duration extends Message<Duration> {
     if (typeof match[2] == "string") {
       const nanosStr = match[2] + "0".repeat(9 - match[2].length);
       this.nanos = parseInt(nanosStr);
-      if (longSeconds < protoInt64.zero) {
+      if (longSeconds < protoInt64.zero || match[1] === '-0') {
```

Comment:
I don't understand it :sweat_smile: Can you explain?

Reply:
Sure.  The regex above, if I'm following correctly, stores the whole number part in `match[1]` and the fractional part in `match[2]`.  In the case of a duration that is > -1 and < 0, then the whole number part (`match[1]`) will be `-0`.  

`BigInt` does not support the notion of a signed zero value (i.e. -0), so when we pass this value to `protoInt64.parse` the returned value is simply 0.  If we simply test `longSeconds` to be < `protoInt64.zero`, that won't work.  We also obviously can't check for this to be `<=`.  So we need a way to know if the whole number part is a string of `-0`.  

We could also test `longSeconds` to equal -0, but that seems error-prone to me over just evaluating an explicit string of `'-0'`.

Reply:
That works too.  Probably a little easier to read.  I just got a bit nervous with dealing with -0 as a number and it felt safer to compare a string.

Reply:
Good call, but thankfully Object.is has [well-defined behaviour](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#description).